beacon-chain/rpc/attester_server.go

```diff
@@ -46,22 +46,25 @@ func (as *AttesterServer) AttestationDataAtSlot(ctx context.Context, req *pb.Att
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve chain head: %v", err)
 	}
-	blockRoot, err := hashutil.HashBeaconBlock(head)
+	headRoot, err := hashutil.HashBeaconBlock(head)
 	if err != nil {
 		return nil, fmt.Errorf("could not tree hash beacon block: %v", err)
 	}
-	beaconState, err := as.beaconDB.State(ctx)
+
+	// Let head state be the state of head block processed through empty slots up to assigned slot.
+	headState, err := as.beaconDB.HistoricalStateFromSlot(ctx, head.Slot)
```

Comment:
Please use `as.beaconDB.State(ctx)` and have an assertion that these slots are the same.

Reply:
We also don't need the assertion, from the spec:
`Let `head_state` be the state of `head_block` processed through any empty slots up to the assigned slot.`

What we need, what I added is this:
```go
	for headState.Slot < req.Slot {
		headState, err = state.ExecuteStateTransition(
			ctx, headState, nil /* block */, headRoot, state.DefaultConfig(),
		)
		if err != nil {
			return nil, fmt.Errorf("could not execute head transition: %v", err)
		}
	}
```